### PR TITLE
[FEATURE] Throw helpful exception if file path does not exist.

### PR DESCRIPTION
**Pull request by smehringer**
This throw was lost in the refactoring

### CODE REVIEW

File: include/chopper/sketch/check_filenames.hpp [hunk 1]
```diff
@@ -34,6 +34,9 @@ inline void check_filenames(std::vector<std::string> const & filenames, configur
 
     for (auto const & filename : filenames)
     {
+        if (!std::filesystem::exists(filename))
+            throw std::invalid_argument{"File " + filename + " does not exist!"};
```
Comment (eseiler) [inline include/chopper/sketch/check_filenames.hpp:38]: You can search for `CHOPPER_WORKAROUND_GCC_BOGUS_MEMCPY` and do the same here

File: include/chopper/sketch/check_filenames.hpp [hunk 2]
```diff
@@ -6,6 +6,17 @@
 
 #include <chopper/configuration.hpp>
 
+/*!\brief Workaround bogus memcpy errors in GCC 12.1 and 12.2. (Wrestrict and Wstringop-overflow)
+ * \see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105545
+ */
+#ifndef CHOPPER_WORKAROUND_GCC_BOGUS_MEMCPY
+#    if defined(__GNUC__) && !defined(__llvm__) && !defined(__INTEL_COMPILER) && (__GNUC__ == 12 && __GNUC_MINOR__ < 3)
```
Comment (eseiler) [inline include/chopper/sketch/check_filenames.hpp:13]:
```suggestion
#    if defined(__GNUC__) && !defined(__llvm__) && !defined(__INTEL_COMPILER) && __GNUC__ == 12
```

**Comment by eseiler** [comment]
The check_filename_test needs some adaptations; but seems to work otherwise

**Comment by smehringer** [comment]
> The check_filename_test needs some adaptations; but seems to work otherwise

Locally the test builds and runs fine. 

CI still complains about the bogus memcpy command somewhere again. Could you check? I'm compiling locally with g++11.3 currently. I'm having troubles upgrading gcc12.1 to a newer version.

**Comment by eseiler** [comment]
> > The check_filename_test needs some adaptations; but seems to work otherwise
> 
> Locally the test builds and runs fine. 

But all the ci (except the one with the bogus error) error on this test. The expects no throw, and now the existence of the files is checked, so the placeholder names in the tests don't work anymore.

I'll check the bogus one later.

**Comment by smehringer** [comment]
You are right, I only checked the gcc12 tests :see_no_evil:

**Pull request merged**